helix-term/src/commands/lsp.rs

```diff
@@ -657,9 +657,20 @@ pub fn hover(cx: &mut Context) {
     );
 }
 pub fn rename_symbol(cx: &mut Context) {
-    ui::prompt(
+    let (view, doc) = current_ref!(cx.editor);
+    let primary_selection = doc
+        .selection(view.id)
+        .primary()
+        .fragment(doc.text().slice(..))
+        .to_string();
+    ui::prompt_with_input(
         cx,
         "rename-to:".into(),
+        if primary_selection.len() > 1 {
+            primary_selection
+        } else {
+            String::new()
+        },
```

Comment:
I think for an initial implementation we can use the word under the cursor as the prefilled text and add support for `textDocument/prepareRename` later. Current word under cursor can be obtained using the [word textobject](https://github.com/helix-editor/helix/blob/d06800f1dd47315f5738c529d40d1e0952ee2115/helix-core/src/textobject.rs#L72).

Reply:
done